manifest/manifestlist/manifestlist_test.go

```diff
@@ -109,3 +110,105 @@ func TestManifestList(t *testing.T) {
 		}
 	}
 }
+
+// TODO (mikebrow): add annotations on the index and individual manifests
+var expectedOCIImageIndexSerialization = []byte(`{
```

Comment:
I don't think it's a given that docker/distribution has to support every single OCI-specification compliant image layout - in particular, there are different use cases for the image-index. For example, by adding org.opencontainers.image.ref.name to the different images, you can logically pack multiple "tags" together into a single image layout. This would make sense in some contexts, but what would it mean to push such an image index to a tag? What would it mean to 'docker pull' from that tag?

@jonjohnsonjr - do you have a particular use case in mind?

(Also, this PR doesn't have to be the be-and-and-end-all of OCI support!)